agent/agentlocal/agent_local.go

```diff
@@ -326,3 +336,50 @@ func (s *Server) runJSONServer(ctx context.Context, grpcAddress string) {
 var (
 	_ agentlocalpb.AgentLocalServer = (*Server)(nil)
 )
+
+// addData add data to zip file
+func addData(zipW *zip.Writer, name string, data []byte) {
+	f, err := zipW.Create(name)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_, err = f.Write(data)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// ZipLogs Handle function for generate zip file with logs.
+func (s *Server) ZipLogs(w http.ResponseWriter, r *http.Request) {
+	buf := &bytes.Buffer{}
+	writer := zip.NewWriter(buf)
+	b := &bytes.Buffer{}
+	for _, serverLog := range s.ringLogs.GetLogs() {
+		_, err := b.WriteString(serverLog)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+	addData(writer, serverZipFile, b.Bytes())
+
+	for id, logs := range s.supervisor.AgentsLogs() {
+		b := &bytes.Buffer{}
+		for _, l := range logs {
+			_, err := b.WriteString(l + "\n")
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+		addData(writer, fmt.Sprintf("%s.txt", id), b.Bytes())
+	}
+	err := writer.Close()
+	if err != nil {
+		log.Fatal(err)
+	}
+	w.Header().Set("Content-Type", "application/zip")
+	w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s.zip\"", "logs"))
+	_, err = w.Write(buf.Bytes())
+	if err != nil {
+		log.Fatal(err)
```

Comment:
@BupycHuk i think this is very bad idea stop whole pmm-agent, what do you think?

Reply:
Did discuss with @BupycHuk